radis/lbl/factory.py

```diff
@@ -799,6 +804,9 @@ def eq_spectrum(
         if verbose:
             print("Spectrum calculated in {0:.2f}s".format(t))
 
+        if (id != "") and (s.lines is not None):
+            s.lines.attrs["id"] = id
+
```

Comment:
This is not needed, and pollutes the code here and above. 
I think you needed it to fix the line survey examples ?

Line survey is a secondary feature so we shouldn't sacrifice code clarity for it. But I think you can  : 
- get the "id" in the line survey by using the Spectrum conditions.
- or (better) : fill the attributes in `lines` by updating the get_lines() function  (L740)